pkg/dexec/borrowed_exec.go

```diff
@@ -125,6 +127,47 @@ func (w *prefixSuffixSaver) Write(p []byte) (n int, err error) {
 	return lenp, nil
 }
 
+// Capture runs a command with the supplied input and captures the
+// output as a string.
+func (c *Cmd) Capture(stdin io.Reader) (output string, err error) {
+	c.Stdin = stdin
+	out := strings.Builder{}
+	c.Stdout = &out
+	err = c.Run()
+	output = out.String()
+	return
+}
+
+// MustCapture is like Capture, but panics if there is an error.
+func (c *Cmd) MustCapture(stdin io.Reader) (output string) {
+	output, err := c.Capture(stdin)
+	if err != nil {
+		panic(err)
+	}
+	return output
+}
+
+// CaptureErr runs a command with the supplied input and captures
+// stdout and stderr as a string.
+func (c *Cmd) CaptureErr(stdin io.Reader) (output string, err error) {
+	c.Stdin = stdin
+	out := strings.Builder{}
+	c.Stdout = &out
+	c.Stderr = &out
+	err = c.Run()
```

Comment:
is it safe to give the same string builder pointer for stdout and stderr? thinking about races and weird interleaving

Reply:
This is a quote from the exec packages godoc:

    // If Stdout and Stderr are the same writer, and have a type that can
    // be compared with ==, at most one goroutine at a time will call Write.

Reply:
cool. that's convenient.